tf2onnx/onnx_opset/tensor.py

```diff
@@ -1097,36 +1098,76 @@ def version_1(cls, ctx, node, **kwargs):
         reorganize_node = ctx.make_node(node.type, trans1.output, attr={"blocksize": blocksize[0]})
         trans2 = ctx.make_node("Transpose", reorganize_node.output, {"perm": [1, 2, 3, 0]})
 
-        # implement crop logic, the data format is NHWC
-        slice_axis = [1, 2]
-        top, bottom = crops[0]
-        left, right = crops[1]
-        starts = [top, left]
-        ends = []
-        for end in [bottom, right]:
-            if end != 0:
-                ends.append(-end)
-            else:
-                ends.append(np.iinfo(np.int32).max)
-
-        attr = {"axes": slice_axis, "ends": ends, "starts": starts}
-        inputs_map = {"data": trans2.output[0], **attr}
         dtypes = node.output_dtypes
         shapes = node.output_shapes
 
-        if len(input_shape) == 3:
-            # add a squeeze op to convert output into 3d
-            kwargs = {**inputs_map}
-            ctx.remove_node(node.name)
-            slice1 = GraphBuilder(ctx).make_slice(kwargs)
-            ctx.make_node("Squeeze", [slice1], {"axes": [3]},
-                          outputs=node.output, name=node.name, dtypes=dtypes, shapes=shapes)
+        if ctx.get_node_by_output(node.input[2]).is_const():
+            crops = node.inputs[2].get_tensor_value()
+            # implement crop logic, the data format is NHWC
+            slice_axis = [1, 2]
+            top, bottom = crops[0]
+            left, right = crops[1]
+            starts = [top, left]
+            ends = []
+            for end in [bottom, right]:
+                if end != 0:
+                    ends.append(-end)
+                else:
+                    ends.append(np.iinfo(np.int32).max)
+
+            attr = {"axes": slice_axis, "ends": ends, "starts": starts}
+            inputs_map = {"data": trans2.output[0], **attr}
+
+            if len(input_shape) == 3:
+                # add a squeeze op to convert output into 3d
+                kwargs = {**inputs_map}
+                ctx.remove_node(node.name)
+                slice1 = GraphBuilder(ctx).make_slice(kwargs)
+                ctx.make_node("Squeeze", [slice1], {"axes": [3]},
+                              outputs=node.output, name=node.name, dtypes=dtypes, shapes=shapes)
+            else:
+                kwargs = {**inputs_map, "outputs": node.output}
```

Comment:
since two branch both need this step, could you move it out so that we only call make_slice once.